lib/go/thrift/simple_json_protocol.go

```diff
@@ -856,30 +853,6 @@ func (p *TSimpleJSONProtocol) ParseStringBody() (string, error) {
 	return v, nil
 }
 
-func (p *TSimpleJSONProtocol) ParseQuotedStringBody() (string, error) {
-	line, err := p.reader.ReadString(JSON_QUOTE)
-	if err != nil {
-		return "", NewTProtocolException(err)
-	}
-	l := len(line)
-	// count number of escapes to see if we need to keep going
-	i := 1
-	for ; i < l; i++ {
-		if line[l-i-1] != '\\' {
-			break
-		}
-	}
-	if i&0x01 == 1 {
-		return line, nil
-	}
-	s, err := p.ParseQuotedStringBody()
-	if err != nil {
-		return "", NewTProtocolException(err)
-	}
-	v := line + s
```

Comment:
my condolences for having to actually read and understand the invariants here! (Thinking about the well-documented `if i&0x01 == 1`, etc)